mitodl/course-search-utils
# Upgrade React to v18

### What are the relevant tickets?





Initially adding Remixicon to import icons directly (in relation to "Search input in filter facet (Topic/Department) formatting issues" in https://github.com/mitodl/hq/issues/5395).

### Description (What does it do?)


We are seeing peer dependency errors when running `npm install` while publishing.

- Upgrades React to v18 (meets peer dependency range for Next.js)
- Upgrades @testing-library/react
- Removes unused dependencies
- Fixes scroll on navigate with Next.js router (page was scrolling to top on filter selection)


### How can this be tested?


Dependencies should install with `npm i` (yarn used on project, but npm is used in the odl-release-bot publish job) and package should build with `yarn prepack`
